### PR TITLE
add module cloudformation_exports

### PR DESCRIPTION
**Pull request by mmoyle**
##### SUMMARY
New module to read CloudFormation Exports

##### COMPONENT NAME
cloudformation_exports

### CODE REVIEW

File: test/integration/targets/cloudformation_exports/tasks/main.yml
Comment (jillr) [inline test/integration/targets/cloudformation_exports/tasks/main.yml]: Please remove the debug task here

File: test/integration/targets/cloudformation_exports/tasks/main.yml
Comment (jillr) [inline test/integration/targets/cloudformation_exports/tasks/main.yml]: We should try to assert some additional things about the export_value, like `export_value['Name'] is "Set By CF Exports"` and the length/number of exports.

File: lib/ansible/modules/cloud/amazon/cloudformation_exports.py
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/cloudformation_exports.py]:
```suggestion
- debug: 
    msg: "{{ cf_exports }}"
```
We generally prefer this syntax.

File: lib/ansible/modules/cloud/amazon/cloudformation_exports.py
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/cloudformation_exports.py]: Please just import the exceptions that you need rather than the full library.

File: lib/ansible/modules/cloud/amazon/cloudformation_exports.py
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/cloudformation_exports.py]:
```suggestion
```
We don't need to return the credentials back out in the result.

File: lib/ansible/modules/cloud/amazon/cloudformation_exports.py
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/cloudformation_exports.py]:
```suggestion
```
Nothing ever gets added to message, if the key isn't used it doesn't need to be returned.

**Review by jillr — COMMENTED** [review]
Thanks for this new module!
Could you please add the module name to `lib/ansible/config/module_defaults.yml` as part of "group/aws", and `cloudformation:ListExports` to `hacking/aws_config/testing_policies/devops-policy.json`?  I'll try to give these tests a run through later this week.

**Comment by tremble** [comment]
@mmoyle When the tests finish ansibot's going to get grumpy about the 'merge' commit.  Please **rebase** rather than merging devel into your branch.

**Comment by mmoyle** [comment]
Hi @jillr ! 

Thank you for the review! I am still getting the testing error

```
01:35     "msg": "Unable to locate credentials", 
01:35     "resource_actions": [
01:35         "cloudformation:ListExports"
01:35     ]
```

File: test/integration/targets/cloudformation_exports/tasks/main.yml
Comment (tremble) [inline test/integration/targets/cloudformation_exports/tasks/main.yml]:
```suggestion
          - export_items | length == 1
```

"that" is kinda like "when" you don't need the wrapping '{{ }}'

**Review by tremble — CHANGES_REQUESTED** [review]

**Comment by jillr** [comment]
Thanks @mmoyle, I haven't deployed your policy changes to aws-terminator yet, when I do that I'll merge that PR and rerun the tests here (it's a straightforward change, looks like it should be fine as-is).  I generally do these once a week on Thursdays.

**Comment by mmoyle** [comment]
> @mmoyle When the tests finish ansibot's going to get grumpy about the 'merge' commit. Please **rebase** rather than merging devel into your branch.

Thank you for the info. I hopefully cleanly removed the merge. I will leave it alone unless we see a merge conflict with devel.

File: test/integration/targets/cloudformation_exports/tasks/main.yml
Comment (jillr) [inline test/integration/targets/cloudformation_exports/tasks/main.yml]: This is actually `cf-export-cf-exports-param`, due to https://github.com/ansible/ansible/blob/a4a4c4817d0c8d33ddf8edfd6a0f82c4a746c63d/test/integration/targets/cloudformation_exports/files/test_stack.yml#L24.

File: test/integration/targets/cloudformation_exports/tasks/main.yml
Comment (jillr) [inline test/integration/targets/cloudformation_exports/tasks/main.yml]:
```suggestion
        export_items: "{{ exports_result['export_items'] }}"
```
I think here you just want export_items, to match up to the assertions being made.

**Review by jillr — COMMENTED** [review]
A couple of notes from testing, this is coming along well!

**Comment by mmoyle** [comment]
btw @jillr, I just got the integration tests to run locally. But I had to place `cloud-config-aws.ini` in two places. In addition, ansible-test could not find the file if I created a `cloud-config-aws.yml`.

I placed the files in
```
./test/integration/cloud-config-aws.ini
./test/lib/ansible_test/config/cloud-config-aws.ini
```
I ran it as 
```
$ ansible-test integration --docker -v cloudformation_exports
```

**Review by tremble — CHANGES_REQUESTED** [review]
The code looks good, but I'm going to be annoying and point out that this module should probably be cloudformation_exports_info (or even just an addition to cloudformation_info)

**Comment by jillr** [comment]
tremble is entirely correct about the module naming - my apologies for the oversight @mmoyle.  I've deployed the CI changes so we should be good after this one last thing.

**Comment by jillr** [comment]
This looks ready to go - I'm going to hold off on merging it for a couple days however as there's some additional work I will need to do in another repo at the time it merges due to our upcoming migration to [collections](https://github.com/ansible-collections/overview/blob/master/README.rst) and we have a few other open PRs for new modules I would like to do all at the same time. So this is all set, you're just waiting on me to hit the merge button now.  Thanks!

**Pull request merged**
